cg/utils/files.py

```diff
@@ -27,3 +31,14 @@ def get_files_matching_pattern(directory: Path, pattern: str) -> List[Path]:
 def get_all_files_in_dir(base_path: Path) -> Set[Path]:
     """Get a set of all files relative to the given base path."""
     return {file.relative_to(base_path) for file in base_path.rglob("*") if file.is_file()}
+
+
+def rename_file(file_path: Path, renamed_file_path: Path) -> None:
+    """Rename the given fastq file path."""
+    if renamed_file_path.exists():
```

Comment:
This might be a single responsibility violation, consider moving the existence check to before this function is called in the main logic